openfisca_uk/variables/benunit/benefits/computed.py

```diff
@@ -0,0 +1,348 @@
+                - benunit("child_working_tax_credit_reduction", period),
+            )
+            / 52
+        )
+
+
+class child_tax_credit(Variable):
+    value_type = float
+    entity = BenUnit
+    label = u"Amount of Working Tax Credit received per week"
+    definition_period = ETERNITY
+
+    def formula(benunit, period, parameters):
+        random = np.random.rand(*benunit("is_single", period).shape)
+        takeup = random < 0.84
+        reduction_left = takeup * min_(
+            0,
+            benunit("working_tax_credit_pre_means_test", period)
+            - benunit("child_working_tax_credit_reduction", period),
+        )
+        return (
+            max_(
+                0,
+                benunit("child_tax_credit_pre_means_test", period)
+                + reduction_left,
+            )
+            / 52
+        )
+
+
+class working_tax_credit_pre_means_test(Variable):
+    value_type = float
+    entity = BenUnit
+    label = (
+        u"Working Tax Credit amount received per year, before means testing"
+    )
+    definition_period = ETERNITY
+
+    def formula(benunit, period, parameters):
+        hours_worked = benunit.sum(benunit.members("hours_worked", period))
+        eligible = (
+            benunit("is_single", period)
+            * (
+                hours_worked
+                >= parameters(
+                    period
+                ).benefits.working_tax_credit.hours_requirement_single
+            )
+            + benunit("is_couple", period)
+            * (
+                hours_worked
+                >= parameters(
+                    period
+                ).benefits.working_tax_credit.hours_requirement_couple
+            )
+            + benunit("is_lone_parent", period)
+            * (
+                hours_worked
+                >= parameters(
+                    period
+                ).benefits.working_tax_credit.hours_requirement_lone_parent
+            )
+        )
+        amount = (
+            parameters(period).benefits.working_tax_credit.amount_basic
+            + (
+                hours_worked
+                >= parameters(
+                    period
+                ).benefits.working_tax_credit.hours_requirement_single
+            )
+            * parameters(period).benefits.working_tax_credit.amount_worker
+            + benunit("is_couple", period)
+            * parameters(period).benefits.working_tax_credit.amount_couple
+            + benunit("is_lone_parent", period)
+            * parameters(period).benefits.working_tax_credit.amount_lone_parent
+        )
+        return amount * eligible
+
+
+class JSA_contributory(Variable):
+    value_type = float
+    entity = BenUnit
+    label = u"JSA (contributory) amount received per week"
+    definition_period = ETERNITY
+
+    def formula(benunit, period, parameters):
+        age = benunit("younger_adult_age", period)
+        is_couple = benunit.nb_persons(BenUnit.ADULT) == 2
```

Comment:
is this a sufficient condition for this? what if an elderly parent is in the benunit?

Reply:
Ok, so there's a mistake here in the contributory JSA in that it seems it actually doesn't have the different rates - will change this. But I think the is_couple condition (if I understand the question right) is correct - a benefit unit can only be a single person or a couple, plus any dependent children so the benefit unit is a couple if and only if the number of adults is exactly 2.

Reply:
This kind of ambiguity between "family" and "benefit unit" (nuclear family) is why I changed the references to family to benunit - hopefully that clears it up a bit.

Reply:
I think in that case yes, you'd have:

- Benefit unit: 2 parents and the 16-year old dependent child
- Benefit unit: 25-year old child
- Benefit unit: elderly person